src/main/java/com/akathist/maven/plugins/launch4j/Launch4jMojo.java

```diff
@@ -929,9 +947,62 @@ private String getLaunch4jVersion() throws MojoExecutionException {
      */
     private boolean skipExecution() {
         getLog().debug("skip = " + this.skip);
-        getLog().debug("skipLaunch4j = " + System.getProperty("skipLaunch4j"));
-        return skip || System.getProperty("skipLaunch4j") != null;
+	String sysProp = System.getProperty("skipLaunch4j");
+        getLog().debug("skipLaunch4j = " + sysProp);
+
+	return skip || (sysProp != null && !sysProp.equalsIgnoreCase("false"));
+	
     }
+    
+    private boolean skipIfNativeImageParam() {
+        getLog().debug("skipIfNativeImage = " + this.skipIfNativeImage);
+	String sysProp = System.getProperty("skipLaunch4jIfNativeImage");
+        getLog().debug("skipLaunch4jIfNativeImage = " + sysProp);
+	
+        return skipIfNativeImage || (sysProp != null && !sysProp.equalsIgnoreCase("false"));
+    }
+    
+   private boolean isNativeImageGenerationActive() {
+	String graalKey = "org.graalvm.buildtools:native-maven-plugin";
+	Log log = getLog();
+	MavenProject curProj = session.getCurrentProject();
+	Build build = curProj.getBuild();
+	String packaging = curProj.getPackaging();
+
+	if (packaging.equalsIgnoreCase("native-image")) {
+	    log.info("Current project's packaging is set to native-image, skipping.");
+	    return true;
+	}
+
+	// check if there's a Graal plugin and if it has any enabled compile executions
+	Plugin graalPlugin = build.getPluginsAsMap().get(graalKey);
+
+	// check if 1) there's a goal with "compile", 2) with no skip params, 3) and no skip sysprops
+	if (graalPlugin != null) {
+            for (PluginExecution pe : graalPlugin.getExecutions()) {
+	        String goalsStr = pe.getGoals().toString().toLowerCase();
+	        String confStr = pe.getConfiguration().toString().toLowerCase();
+	    
+	        if (!goalsStr.contains("compile") || pe.getPhase() == null) {
+                    continue;
+	        }
+		if (confStr.contains("<skip>true") || confStr.contains("<skipNativeBuild>true"))
```

Comment:
There is no way to detect that without parsing the whole configuration? What if there be `<skip>true` but not related to Graal plugin, eg. to skip tests?

Reply:
It's not parsing the whole configuration, only the fragment from a specific execution, for example:

confStr=>>><?xml version="1.0" encoding="utf-8"?>
<configuration>
  <buildargs combine.children="append">
    <buildarg>--no-fallback</buildarg>
    <buildarg>--no-fallback</buildarg>
    <buildarg>--no-fallback</buildarg></buildargs>
  <imagename>xxxxxxxxxx</imagename>
  <mainclass>com.example.xxxxxxxxxx</mainclass>
  <metadatarepository>
    <enabled>true</enabled>
  </metadatarepository>
  <isdetectionenabled>true</isdetectionenabled>
</configuration><<<

Note it's not checking the file-pom, only xml recreated from already parsed and merged effective pom. Also it ignores executions that don't have "compile" word in them, so it shouldn't interfere with things like metadata gathering / merging.
I'm certain there will be cases when it will fail, but it should suffice for 90%+ of typical format+compile+shade+package projects. Certainly for the ones in which the user just copy-pastes and slightly modifies the parts relevant to Graal and Launch4j.

p.s. I have no idea why "--no-fallback" is repeated, it's just a basic Micronaut project, using their generator, with simplest Launch4j fragment added.